sds/src/scanner/mod.rs

```diff
@@ -53,6 +54,8 @@ pub trait CompiledRuleTrait: Send + Sync {
         content: &str,
         path: &Path,
         caches: &mut CachePoolGuard<'_>,
+        rule_idx: usize,
+        cached_string_matches_per_rule_idx: &mut HashMap<usize, AHashSet<String>>,
```

Comment:
If only the entry of the cache associated to the `rule_idx` will be used, would it make sense to only keep this entry in the `get_string_matches` method, in other words:
```suggestion
        cached_string_matches: &mut AHashSet<String>,
```